lib/HLSL/HLOperationLower.cpp

```diff
@@ -600,18 +600,30 @@ Value *TrivialBarrier(CallInst *CI, IntrinsicOp IOP, OP::OpCode opcode,
 Value *TranslateD3DColorToUByte4(CallInst *CI, IntrinsicOp IOP,
                                  OP::OpCode opcode,
                                  HLOperationLowerHelper &helper,  HLObjectOperationLowerHelper *pObjHelper, bool &Translated) {
-  hlsl::OP *hlslOP = &helper.hlslOP;
   IRBuilder<> Builder(CI);
   Value *val = CI->getArgOperand(HLOperandIndex::kUnaryOpSrc0Idx);
   Type *Ty = val->getType();
 
-  Constant *toByteConst = ConstantFP::get(Ty->getScalarType(), 255);
-  if (Ty != Ty->getScalarType()) {
-    toByteConst =
-        ConstantVector::getSplat(Ty->getVectorNumElements(), toByteConst);
+  bool isFXCCompatMode = CI->getModule()->GetHLModule().GetHLOptions().bFXCCompatMode;
+  // Use literal 255.001953 for scaling when FXC compat mode is enabled as
+  // FXC uses this value instead of 255 for some uknown reason.
+  static const float scalingVal = isFXCCompatMode ? 255.001953 : 255.0;
```

Comment:
Can we just use the FXC magic number everywhere? Every time we have a conditional on a compilation option, we increase the test surface and interplay between slightly different behaviors. I think this one is not worth the complexity.

Also won't this complain about implicit double-to-float conversions?

Reply:
And `static` here is weird, I'm surprised it compiles!

Reply:
With your reference to stackoverflow thread on FXC magic number, it seems there is some reasoning behind it. So I will use it everywhere.